tuftool/src/update.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::build_targets;
+use crate::datetime::parse_datetime;
+use crate::error::{self, Result};
+use crate::source::parse_key_source;
+use chrono::{DateTime, Utc};
+use snafu::ResultExt;
+use std::fs::File;
+use std::num::{NonZeroU64, NonZeroUsize};
+use std::path::PathBuf;
+use structopt::StructOpt;
+use tough::editor::RepositoryEditor;
+use tough::key_source::KeySource;
+use tough::{ExpirationEnforcement, FilesystemTransport, HttpTransport, Limits, Repository};
+use url::Url;
+
+#[derive(Debug, StructOpt)]
+pub(crate) struct UpdateArgs {
+    /// Follow symbolic links in `indir`
+    #[structopt(short = "f", long = "follow")]
+    follow: bool,
+
+    /// Number of target hashing threads to run (default: number of cores)
+    // No default is specified in structopt here. This is because rayon
+    // automatically spawns the same number of threads as cores when any
+    // of its parallel methods are called.
+    #[structopt(short = "j", long = "jobs")]
+    jobs: Option<NonZeroUsize>,
+
+    /// Key files to sign with
+    #[structopt(short = "k", long = "key", required = true, parse(try_from_str = parse_key_source))]
+    keys: Vec<Box<dyn KeySource>>,
```

Comment:
Why is it `keys` and "files" and `Vec` when `parse_key_source` returns a single `Box<dyn KeySource>`?  I feel like I'm missing something, does structopt automatically handle multiple arguments?

Reply:
I followed the lead of the other modules that include this argument. Structopt allows a user to pass [multiple identical arguments](https://docs.rs/structopt/0.3.14/structopt/index.html#type-magic) by parsing the type; `Vec` in this case.

I don't think it will be very common, but using a `Vec` here allows a user to pass multiple `-k` arguments with different key sources.